job.go

```diff
@@ -45,46 +53,55 @@ func (j *Job) shouldRun() bool {
 }
 
 //Run the job and immediately reschedule it
-func (j *Job) run() (result []reflect.Value, err error) {
+func (j *Job) run() ([]reflect.Value, error) {
 	if j.lock {
 		if locker == nil {
-			err = fmt.Errorf("trying to lock %s with nil locker", j.jobFunc)
-			return
+			return nil, fmt.Errorf("trying to lock %s with nil locker", j.jobFunc)
 		}
 		key := getFunctionKey(j.jobFunc)
 
-		if ok, err := locker.Lock(key); err != nil || !ok {
-			return nil, err
-		}
-
-		defer func() {
-			if e := locker.Unlock(key); e != nil {
-				err = e
-			}
-		}()
+		locker.Lock(key)
```

Comment:
will change our locker signature in the future to remove errors just like Sync.Locker